### ISSUE TITLE
parent mode is c-mode in `define-derived-mode`

### ISSUE THREAD

**Issue by tsuu32** [ethereum/emacs-solidity#49]
I think this is problematic.

I use
```
(add-hook 'c-mode-hook #'irony-mode)
```
in init file,  so solidity-mode enable irony-mode (undesirable).

Solidity-mode should be
```
(define-derived-mode solidity-mode prog-mode "solidity"
```
?

**Comment by LefterisJP** [comment]
Hello @tsuu32! 

Thank you for trying out emacs-solidity.

When I first made this language mode it was suggested from various guides to derive from `c-mode` for languages like Solidity which are similar to C in syntax. Example guide seen [here](http://ergoemacs.org/emacs/elisp_syntax_coloring.html).

We can try `prog-mode` to solve your problem but it may need some more work than just switching `c-mode` with `prog-mode` as there may be some characteristics of c-mode that will need to be reconfigured.

I can try it when I find some time, or if you also have time Pull Requests are more than welcome!

**Comment by LefterisJP** [comment]
Fixed by #50. Thank you.
